lib/src/install/config.rs

```diff
@@ -81,28 +89,43 @@ where
 
 impl Mergeable for RootFS {
     /// Apply any values in other, overriding any existing values in `self`.
-    fn merge(&mut self, other: Self) {
-        merge_basic(&mut self.fstype, other.fstype)
+    fn merge(&mut self, other: Self, env: &EnvProperties) {
+        merge_basic(&mut self.fstype, other.fstype, env)
     }
 }
 
 impl Mergeable for BasicFilesystems {
     /// Apply any values in other, overriding any existing values in `self`.
-    fn merge(&mut self, other: Self) {
-        self.root.merge(other.root)
+    fn merge(&mut self, other: Self, env: &EnvProperties) {
+        self.root.merge(other.root, env)
     }
 }
 
 impl Mergeable for InstallConfiguration {
     /// Apply any values in other, overriding any existing values in `self`.
-    fn merge(&mut self, other: Self) {
-        merge_basic(&mut self.root_fs_type, other.root_fs_type);
-        merge_basic(&mut self.block, other.block);
-        self.filesystem.merge(other.filesystem);
-        if let Some(other_kargs) = other.kargs {
-            self.kargs
-                .get_or_insert_with(Default::default)
-                .extend(other_kargs)
+    fn merge(&mut self, other: Self, env: &EnvProperties) {
+        // if arch is specified, only merge config if it matches the current arch
+        // if arch is not specified, merge config unconditionally
+        if let Some(other_arch) = other.match_architectures {
+            if other_arch.contains(&env.sys_arch) {
```

Comment:
We can avoid duplicating the processing logic I think via this:

```
if other.match_architectures.map(|a| a.contains(&env.sys_arch)).unwrap_or(true) {
```

Reply:
Oh I didn't know you could do this, very cool :)

Reply:
No worries; I feel like it took me a solid year of writing Rust often to really start to feel like I'd mastered the whole sub-language around combinators.  But it's cool when you get there :smile: